x-pack/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/three_way_diff/comparison_side/subfield.tsx

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import type { DiffableAllFields } from '../../../../../../../common/api/detection_engine';
+import { DiffView } from '../../json_diff/diff_view';
+import { SubfieldHeader } from './subfield_header';
+
+const FIELDS_WITH_SUBFIELDS: Array<keyof DiffableAllFields> = [
+  'data_source',
+  'kql_query',
+  'eql_query',
+  'esql_query',
+  'threat_query',
+  'rule_schedule',
+  'rule_name_override',
+  'timestamp_override',
+  'timeline_template',
+  'building_block',
+  'threshold',
+];
```

Comment:
nit: this could be put in a `constants.ts` file in this folder or in the above folder, we'll likely have other stuff to add to it or reference these at some point in the future (tests, etc)

Reply:
Done!